bpf/flows.c

```diff
@@ -102,4 +25,32 @@ int egress_flow_parse(struct __sk_buff *skb) {
     return flow_monitor(skb, EGRESS);
 }
 
+SEC("tracepoint/skb/kfree_skb")
+int kfree_skb(struct trace_event_raw_kfree_skb *args) {
+    struct sk_buff skb;
+    __builtin_memset(&skb, 0, sizeof(skb));
+
+    bpf_probe_read(&skb, sizeof(struct sk_buff), args->skbaddr);
+    struct sock *sk = skb.sk;
+    enum skb_drop_reason reason = args->reason;
+
+    // SKB_NOT_DROPPED_YET,
+    // SKB_CONSUMED,
+    // SKB_DROP_REASON_NOT_SPECIFIED,
+    if (reason > SKB_DROP_REASON_NOT_SPECIFIED) {
+        return trace_tcp_drop(args, sk, &skb, reason);
+    }
+    return 0;
+}
+
+SEC("tracepoint/net/net_dev_queue")
+int trace_net_packets(struct trace_event_raw_net_dev_template *args) {
+    struct sk_buff skb;
+
+    __builtin_memset(&skb, 0, sizeof(skb));
+    bpf_probe_read(&skb, sizeof(struct sk_buff), args->skbaddr);
+    return trace_dns(&skb);
+}
+
```

Comment:
why did move all sec here instead of keeping it in its own file as it was ? because this file will be empty ? maybe that is ok I think its cleaner to keep the implementation with the ist section and `.c` file is just container for all features

Reply:
yes that's what my thought was, after following your refactoring style I felt that this file was getting empty so I thought we can put all the various hookpoints in the .c file for visibility on which features the agent supports (optional or not doesn't matter) and keep the feature specefic implementation in their respective files.

If this is not okay or a clear point I am happy to move things back like before.

Reply:
yeah I would move the SEC to each feature header even TC to the new hdr u added and just add a comment in the .c file stating this is a container file for all the different eBPF hooks the netobserv agent supports and we do so because bpf2go limit us to use only single `.c` file to generate go structs

Reply:
Took care of the comment. Will be visible in the latest commit.